File: native-with-db-server/app/src/main/java/ubb/barcoaie/lab5_ma/activities/Delete/DeleteActivity.java
package ubb.barcoaie.lab5_ma.activities.Delete;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;

import ubb.barcoaie.lab5_ma.R;
import ubb.barcoaie.lab5_ma.VegetableViewModel;

public class DeleteActivity extends AppCompatActivity {
    private VegetableViewModel vegetableViewModel;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_delete);

        RecyclerView recyclerView = findViewById(R.id.recyclerView);
        final VegetableListAdapterDel adapter = new VegetableListAdapterDel(new VegetableListAdapterDel.VegetableDiff());
        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        vegetableViewModel = new ViewModelProvider(this, new ViewModelProvider.Factory() {
            @NonNull
            @Override
            public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
                return (T) new VegetableViewModel(getApplication());
            }
        }).get(VegetableViewModel.class);
        vegetableViewModel.getAllVegetables().observe(this, adapter::submitList);
    }

    @Override
    public void onBackPressed() {
        Intent replyIntent = new Intent();
        setResult(RESULT_OK, replyIntent);
        DeleteActivity.this.finish();
    }
}

File: native/app/src/main/java/ubb/barcoaie/lab2_ma/Activities/DeleteActivity.kt
package ubb.barcoaie.lab2_ma.Activities

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.GestureDetector
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.ListView
import androidx.core.view.get
import kotlinx.android.synthetic.main.activity_delete.*
import ubb.barcoaie.lab2_ma.R
import ubb.barcoaie.lab2_ma.Repo.VegetableRepoImpl

class DeleteActivity : AppCompatActivity() {
    var repo = VegetableRepoImpl()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_delete)

        val deleteBtn: Button = findViewById(R.id.delete_btn);

        repo = intent.extras?.get("Repo") as VegetableRepoImpl

        deleteBtn.setOnClickListener {
            val input_name_text = input_name.text.toString()
            repo.delVegetable(input_name_text)
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        val intent = Intent()
        setResult(RESULT_OK, intent)
        intent.putExtra("Repo", repo)
        this.finish()
    }
}